components/Helpers/src/ThemeListenerHelperWindow.cs

```diff
@@ -0,0 +1,92 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Win32;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.Win32;
+
+namespace CommunityToolkit.WinUI.HelpersRns;
+
+#if WINAPPSDK
+
+internal class ThemeListenerHelperWindow
+{
+    public delegate void ThemeChangedHandler(ApplicationTheme theme);
+
+    public static ThemeListenerHelperWindow Instance = new();
+    private static string s_className = "ThemeListenerHelperWindow";
+    private IntPtr m_hwnd;
+    public event ThemeChangedHandler? ThemeChanged;
+    delegate Windows.Win32.Foundation.LRESULT WndProc(Windows.Win32.Foundation.HWND hWnd, uint msg, Windows.Win32.Foundation.WPARAM wParam, Windows.Win32.Foundation.LPARAM lParam);
+    private WndProc m_wnd_proc_delegate;
+    private void registerClass()
+    {
+        unsafe
+        {
+            var wcx = new Windows.Win32.UI.WindowsAndMessaging.WNDCLASSEXW();
+            wcx.cbSize = (uint)Marshal.SizeOf(wcx);
```

Comment:
This is not AOT safe. You need to configure CsWin32 in blittable mode ([copy this](https://github.com/Sergio0694/ComputeSharp/blob/main/src/ComputeSharp.SourceGenerators/NativeMethods.json))

Reply:
> I am not familiar with this. Is there any doc I can look into?

@HO-COOH CsWin32 uses built-in "runtime" com marshalers by default, which isnt supported by native aot. While you don't have to disable friendly overload, you have to disable marshaling.
CsWin32 has [docs](https://github.com/colejohnson66/CsWin32/tree/patch-1?tab=readme-ov-file#support-for-trimming-aot-andor-disabling-the-runtime-marshaler) for this.

![IMG_0431](https://github.com/user-attachments/assets/06b60a28-5a70-4eaf-9da3-62795b0aafb9)